src/ApiBundle/Controller/AuthController.php

```diff
@@ -503,12 +440,30 @@ public function editProfileAction()
 
         $data = $request->request->all();
 
-        $this->handleKeyUsername($user, $request);
-        $this->handleKeyEmail($user, $request);
-        $this->handleKeyFirstname($user, $request);
-        $this->handleKeyLastname($user, $request);
-        $this->handleKeyDob($user, $request);
+        $user->setFirstname(array_key_exists('firstname', $data) ? $data['firstname'] : $user->getFirstname() );
+        $user->setLastname(array_key_exists('lastname', $data) ? $data['lastname'] : $user->getLastname() );
+        $user->setDob( array_key_exists('dob', $data) ? $data['dob'] : $user->getDob() );
+
+        // Validate user data
+        $validator = $this->get('validator');
+        $errors = $validator->validate($user, null, array('profile_edit'));
+
+        if (count($errors) > 0) {
+            $errorArray = [];
+            foreach ($errors as $error) {
+                $constraint = $error->getConstraint();
+                $errorItem = array(
+                                    "code" => 400,
+                                    "error" =>  "Bad Request",
+                                    "error_description" => $error->getMessage(),
```

Comment:
I'm not sure if @mshariq-nerd @ajabble suggest, to keep `error_type` as well? and  in `error_details` we should add 'field_name' = "FIRST_NAME", 'field_name' = "EMAIL" .... as well